### PR TITLE
Remove support for editing shared addresses while request to join is pending approval

### PR DESCRIPTION
**Pull request by ilmotta**
### Summary

This PR completely removes support for editing shared addresses while a user's request to join is pending approval.

The reason for the removal is that, in status-go, the control node that will process the message `protobuf.ApplicationMetadataMessage_COMMUNITY_EDIT_SHARED_ADDRESSES` verifies the user is already a member of the community, but a user waiting for approval to join is not yet a member. Removing this check doesn't sound like a good idea and we want to avoid bugs/complications for the next release. The feature isn't working anyway, so the purpose of this PR is also to cleanup code that doesn't work and won't be re-done anytime soon.

#### Areas that may be impacted

In theory none because we (devs & QAs) know enabling the feature flag doesn't cause regressions in other parts of the app.

### Steps to test

Dear QAs, in this PR, besides removing support for editing addresses while requests to join are pending, I'm also enabling by default the feature flag to edit addresses since the feature seems to be stable enough that the flag is unjustified now. This should facilitate testing on your part. If you have already tested editing shared addresses (I can't be sure now), then I simple smoke test should be enough because I didn't change the behavior for anything except requests to join in the pending state, which I'm sure you couldn't test because it never worked.

~Before merging this PR, and after you approve from a QA standpoint, I will remove the flag entirely from the code (it's quick & safe).~ Edit: We will only remove the flag once other unrelated problems are solved, which will allow QAs to thoroughly test the feature.

status: ready

### CODE REVIEW

File: src/status_im/feature_flags.cljs
Comment (ilmotta) [inline src/status_im/feature_flags.cljs]: FYI: This flag will be removed and all related code before this PR merged and after QAs approve it.

File: src/status_im/subs/community/account_selection.cljs
Comment (ibrkhalil) [inline src/status_im/subs/community/account_selection.cljs]: Since it's only a single arg now, Do we need double vectors?

Comment (ilmotta) [reply src/status_im/subs/community/account_selection.cljs]: You are correct, we indeed don't need the vector. I dislike how re-frame is inconsistent at times. Besides your suggestion, I can also remove the vector from line 33, but keep the destructuring as is and it still works...

Thanks @ibrkhalil

Comment (ibrkhalil) [reply src/status_im/subs/community/account_selection.cljs]: I remember @flexsurfer telling me it causes memory leaks sometimes.

**Review by ibrkhalil — APPROVED** [review]
Other than the sub issue, As always great work!

File: src/status_im/subs/community/account_selection.cljs
Comment (ilmotta) [reply src/status_im/subs/community/account_selection.cljs]: I found a few other subs in `develop` (`:communities/community-color`, `:communities/community-outro-message`, `:communities/community-members`, etc) that have the extra/unnecessary vector in the destructuring part, but I don't understand how that could cause a leak.

When there's only one input signal, re-frame supports the first argument from the sub handler to be either a vector or a single argument. If we truly identified a leak due to this API detail, then we should definitely report it as a bug.

Comment (ilmotta) [reply src/status_im/subs/community/account_selection.cljs]: Oh @ibrkhalil, I think I remember now what you mean. Sometimes a developer can forget to remove an input signal, even thought it's not used anymore in the handler. In that way, this causes re-frame to do more work than necessary. For example (artificial, not in `develop`):

```clojure
(re-frame/reg-sub :communities/community-joined
 (fn [[_ community-id]]
   [(re-frame/subscribe [:communities/community community-id])
    (re-frame/subscribe [:communities/community-color community-id])])
 (fn [[{:keys [joined]}] _]
   joined))
```

See ya :wave: 🤝

**Review by jo-mut — APPROVED** [review]

**Review by alwx — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by ajayesivan — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hey @ilmotta! Thank you for the PR. 

Here is some update:
During PR testing I have faced some issues related to updating roles and permissions based on selected addresses. For example, channel permissions were not updated based on selected/unselected accounts which are holding required tokens. It turned out that control node functionality is broken on Desktop https://github.com/status-im/status-desktop/issues/14472

As updating of permissions is directly connected to control node from my point of view there is not sense to continue testing of this PR until the above issue is resolved on Desktop side. 

Apologies for forced delay in testing @ilmotta

**Comment by pavloburykh** [comment]
> For example, channel permissions were not updated based on selected/unselected accounts which are holding required tokens.

@ilmotta I will give you an example, so you can check. Maybe this is still mobile related issue and not related to Desktop bug

### ISSUE 1 Selecting / unselecting account addresses during editing does not affect channel permissions

**Preconditions:** 
here are my community permissions: 

1. Anyone who holds MBCOL is eligible to join as Member
2. Anyone who holds CNONE is eligible to View and Post in chan-1

Status Desktop 2024-04-18 17-36-37

My mobile user holds **MBCOL** on ```Membership``` account and **CNONE** on ```chan 2``` account (you can see it on video below the Steps)

**Steps:** 

1. Mobile user opens community
2. Unselect ```chan 2``` account which holds CNONE collectible required to access ```chan-1``` channel
3. Join the community
4. After joining open Editing shared addresses
5. Select ```chan 2``` account which holds CNONE collectible required to access ```chan-1``` channel
6. Confirm changes
7. See if mobile user is able to access chan-1 

**Expected result:** user is able to access chan-1 as he shared ```chan 2``` account which holds CNONE collectible required to access ```chan-1``` channel


**Actual result:** unable to access ```chan-1``` channel as it remains locked. 

[Desktop_control_node.log.zip](https://github.com/status-im/status-mobile/files/15026598/Desktop_control_node.log.zip)
[mobile_user_edited_address.zip](https://github.com/status-im/status-mobile/files/15026599/mobile_user_edited_address.zip)


https://github.com/status-im/status-mobile/assets/97245802/f89c5e07-ccc5-487b-9632-767ad67702b9



@ilmotta there might be explanation, that permissions are updated once an hour on control node side (@cammellos knows better about these details). I have been waiting for a long time with control node being online but at the moment of posting this comment channel permissions have not been updated according to selected account. 

If permissions will be subsequently updated after some more time - then we will need some special Desktop build with updated value of permissions updating frequency, let's say once in 1-5 minutes. Otherwise testing will be painfull.

**Comment by pavloburykh** [comment]
so basically, currently I can test PR from only from UI perspective. But I cannot verify if editing shared addresses in fact applies on BE side due to the problems described above.

**Comment by pavloburykh** [comment]
@ilmotta **UPDATE:**

Desktop [issue](https://github.com/status-im/status-desktop/issues/14472) has been fixed but unfortunately [ISSUE 1](https://github.com/status-im/status-mobile/pull/19623#issuecomment-2064102939) is still reproducible. 

 > If permissions will be subsequently updated after some more time - then we will need some special Desktop build with updated value of permissions updating frequency, let's say once in 1-5 minutes.
 
 permissions have not been updated even after hours, so it seems that it just doesn't work.

**Comment by ilmotta** [comment]
Hey @pavloburykh, thanks a lot for all the comments. I wasn't aware of such problems.

I'm *theoretically* (if I deleted the code correctly that is) only removing code that can't be reached by the user because they were all related to editing shared addresses while having a pending request to join. In that way, there's nothing to test in this PR that's new or that wasn't tested before. Functionally, it should really be identical to `develop`.

But given that we are facing these other problems you mentioned, I agree it's better to stabilize first and make things solid and only then remove the flag in a future PR. OTOH, I think we don't need to block this PR because it's a pretty safe refactor, which I also smoke tested, but it would be valuable if you could smoke test a basic flow too.

So what do you say we:

1. Merge this PR.
2. Continue to iron out the features as usual in other PRs.
3. When we believe we're good to go we create a very simple PR that just removes the flag and then QAs can test again to give the go/no-go.

**Comment by pavloburykh** [comment]
@ilmotta agree. Let's merge it but keep the flag until we figure out issues with permissions. Thank you!

**Pull request merged**
